docs/lib/manual/Flash.md

```diff
@@ -0,0 +1,142 @@
+
+To find out which peripheral is responsible for this fault, you can look in the [peripheral access layer](https://github.com/ARMmbed/mbed-os/blob/e5ba1d2/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/device/MK64F12.h) for your development board. If you search here for `40020000` - peripherals are normally aligned - you find the `FTFE_BASE` address, which is the flash driver for the K64F.
+
+```cpp
+/** Peripheral FTFE base address */
+#define FTFE_BASE                                (0x40020000u)
+```
+
+You can enable this peripheral for the secure box by adding the following definition to the ACL list:
+
+```cpp
+    {FTFE,             sizeof(*FTFE),  UVISOR_TACLDEF_PERIPH},
+```
+
+After recompiling the code, you can now see - over the serial port - that you can successfully write to flash. However, uVisor still halts when trying to read from flash.
+
+```
+***********************************************************
+                    BUS FAULT
+***********************************************************
+
+* Active Box ID: 1
+* FAULT SYNDROME REGISTERS
+
+  CFSR: 0x00008200
+  BFAR: 0x000FF000
+  --> PRECISERR: precise data access.
+
+* MPU FAULT:
+  Slave port:       0
+  Address:          0x000FF000
+  Faulting regions:
+    R00: 0x00000000 0xFFFFFFFF 0x000007D0 0x00000001
+  Master port:      0
+  Error attribute:  Data READ (user mode)
+
+HALT_ERROR(./core/vmpu/src/kinetis/vmpu_kinetis.c#144): Access to restricted resource denied
+```
+
+## Configuring the ACL to read from flash
+
+As you can see from the log message, the MPU prevents reading from address `0x000FF000`. This is the value of the `address` you try to read. The K64F uses a memory-mapped address space to access flash storage, which is mapped from `0x0` to `0x100000`, and you're using a sector size of `0x1000` bytes. To get around this you can add this specific part of the address space to the ACL list. A nice addition is that you can limit boxes to only part of the flash storage. It's not all or nothing.
+
+To allow reading from the last sector of flash storage, add the following definition to the ACL list of the secure box:
+
+```cpp
+    {(void *)0xff000,   SECTOR_SIZE,    UVISOR_TACL_ACCESS | UVISOR_TACL_SHARED},
+```
+
+When you recompile, the program runs to completion, and you can both read and write to flash from your secure box.
+
+## Conclusion
+
+uVisor can be used not just for isolating memory between boxes, or to block usage of certain peripherals. You can also use uVisor to block off reading parts of the flash storage from unauthorized boxes. Unfortunately, because writing to flash goes through the FTFE peripheral, you do not have the same fine-grained control over writing to flash. You can solve this by adding a specific secure box which owns the FTFE peripheral, and use a [public secure entry point](../QUICKSTART.md#Expose-public-secure-entry-points-to-the-secure-box) to control which boxes can write to (specific parts of) flash.
```

Comment:
```diff
- (../QUICKSTART.md#Expose-public-secure-entry-points-to-the-secure-box)
+ (../QUICKSTART.md#expose-public-secure-entry-points-to-the-secure-box)
```

Otherwise, the anchor link doesn't work.

Reply:
Fixed.